src/kakarot/interpreter.cairo

```diff
@@ -753,13 +752,12 @@ namespace Interpreter {
             calldata=calldata,
             calldata_len=calldata_len,
             value=value,
-            gas_price=gas_price,
-            origin=origin,
             parent=cast(0, model.Parent*),
             address=address,
             read_only=FALSE,
             is_create=is_deploy_tx,
             depth=0,
+            env=env,
```

Comment:
I would not put the environment inside the message, as they have no direct relation

Reply:
I also think execute should just receive a `message` and an `environment` and process the execution of the message against that environment, and the preparation of the message should be done before

Reply:
I see things like 
env > message > execution

meaning that when someone triggers a tx, first the env is defined, then within this env, messages a created, and enventually run

not mentionning the fact that putting `env` in `evm` costs more for nothing

Reply:
regarding the processing of the message against the env, the thing is that here the state is not part of the env, so the env is really just external static values

Reply:
hm, ok makes sense in that case